reactnative_mobile/react_app_todo_list/App.tsx

```diff
@@ -30,27 +54,89 @@ export default function App() {
     appState.current = nextAppState;
   };
 
-  useEffect(() => {
-    const subscription = AppState.addEventListener(
-      "change",
-      handleAppStateChange
-    );
-    return () => {
-      subscription.remove();
-    };
-  }, []);
-
-  const handleAddTask = async (text: string) => {
+  const handleAddTask = async (modelObj: TODOModel) => {
     Keyboard.dismiss();
-    setTaskItems([...taskItems, task]);
-    setTask(null);
+    setTask("");
     Statsig.logEvent(TODO_CREATED);
+    addTodoInList(modelObj);
+    fetchTodoList();
+  };
+
+  const addTodoInList = async (modelObj: TODOModel) => {
+    fetch(baseTodoUrl, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        serialNumber: modelObj.serialNumber,
+        task: modelObj.task,
+        completed: modelObj.completed,
+        description: modelObj.description,
+        edited: modelObj.edited,
+        createdDate: modelObj.createdDate,
+        modifiedDate: modelObj.modifiedDate,
+        lastViewed: modelObj.lastViewed,
+      }),
+    })
+      .then((response) => {})
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+
+  const fetchTodoList = async () => {
+    if (todoList.length > 0) {
+      setTodoList([]);
+    }
+    try {
+      const response = await fetch(baseTodoUrl);
+      const json = await response.json();
+      setTodoList(json);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const deleteTask = (item: TODOModel) => {
+    const url = baseTodoUrl + "/" + item.id;
+    fetch(url, {
+      method: "DELETE",
+    })
+      .then((response) => response.json())
```

Comment:
Use async/await rather than promise chaining

Reply:
Will change it.

Reply:
Will change it.